reframe/frontend/executors/__init__.py

```diff
@@ -197,6 +205,21 @@ def _environ_supported(self, check, environ):
         else:
             return ret and check.supports_environ(environ.name)
 
+    def _retry_failed(self, checks, system):
+        while (self._stats.num_failures() and self._current_run <
+                                              self._max_retries):
+            self._current_run += 1
+            self._stats.next_run(self._current_run)
+            check_names_failed = set([t.check.name for t in
+                                      self._stats.tasks_failed(run=-2)])
+            checks_failed = [c for c in checks if c.name in check_names_failed]
```

Comment:
You can rewrite this with a single list comprehension:

```python
            failed_checks = [
                c for c in checks if c.name in
                set(t.check.name for t in self._stats.tasks_failed())
            ]
```

Reply:
I prefer also the name `failed_checks` over the `checks_failed`.

Reply:
Done.